src/resourcetiming-compression.js

```diff
@@ -803,6 +804,23 @@
         return this.optimizeTrie(this.convertToTrie(results), true);
     };
 
+    // eslint-disable-next-line max-len
+    ResourceTimingCompression.HOSTNAME_REGEX = /^(?:(?:(([^:\/#\?]+:)?(?:(?:\/\/)(?:(?:(?:([^:@\/#\?]+)(?:\:([^:@\/#\?]*))?)@)?(([^:\/#\?\]\[]+|\[[^\/\]@#?]+\])(?:\:([0-9]+))?))?)?)?((?:\/?(?:[^\/\?#]+\/+)*)(?:[^\?#]*)))?(\?[^#]+)?)(#.*)?/;
+    ResourceTimingCompression.reverseHostname = function(url) {
+        var hostname = (ResourceTimingCompression.HOSTNAME_REGEX.exec(url) || [])[5];
+        return typeof hostname === "string"
+            ? url.split(hostname).join(ResourceTimingCompression.reverseString(hostname))
+            : url;
+    };
+
+    ResourceTimingCompression.reverseString = function(i) {
+        var l = i.length, o = "";
```

Comment:
Not that we have to hyper-optimize this, but I came across this test:
https://jsperf.com/reverse-string

The split/reverse/join seems to be fastest overall (though I don't think we should update the `String.prototype`)